foam_controlwrapper/tests/test_foam_interface.py

```diff
@@ -0,0 +1,292 @@
+            label += 1
+
+        label = 0
+        for edge in self.mesh.iter_edges():
+            uid = edge.uid
+            self._uuidToFoamLabel[uid] = label
+            self._foamEdgeLabelToUuid[label] = uid
+            label += 1
+
+        label = 0
+        for face in self.mesh.iter_faces():
+            uid = face.uid
+            self._uuidToFoamLabel[uid] = label
+            self._foamFaceLabelToUuid[label] = uid
+            label += 1
+
+        label = 0
+        for cell in self.mesh.iter_cells():
+            uid = cell.uid
+            self._uuidToFoamLabel[uid] = label
+            self._foamCellLabelToUuid[label] = uid
+            label += 1
+
+        # find out boundary patches
+        patchNameFacesMap = OrderedDict()
+        self.facePoints = []
+        for face in self.mesh.iter_faces():
+            if CUBA.LABEL in face.data:
+                boundary = 'boundary' + str(face.data[CUBA.LABEL])
+                if boundary not in patchNameFacesMap:
+                    patchNameFacesMap[boundary] = []
+                patchNameFacesMap[boundary].append(
+                    self._uuidToFoamLabel[face.uid])
+
+            # make compressed list of faces points
+            self.facePoints.append(len(face.points))
+            for puid in face.points:
+                self.facePoints.append(self._uuidToFoamLabel[puid])
+
+        # make points coordinate list
+        self.pointCoordinates = []
+        for point in self.mesh.iter_points():
+            for coord in point.coordinates:
+                self.pointCoordinates.append(coord)
+
+        # make compressed list of cells points
+        self.cellPoints = []
+        for cell in self.mesh.iter_cells():
+            self.cellPoints.append(len(cell.points))
+            for puid in cell.points:
+                self.cellPoints.append(self._uuidToFoamLabel[puid])
+
+        # make patch information
+        self.patchNames = []
+        self.patchFaces = []
+        for patchName in patchNameFacesMap:
+            self.patchNames.append(patchName)
+            self.patchFaces.append(len(patchNameFacesMap[patchName]))
+            for face in patchNameFacesMap[patchName]:
+                self.patchFaces.append(face)
+
+        if not self.patchNames:
+            error_str = 'Could not initialize with mesh  {}. '
+            error_str += 'Mesh has not boundary face definitions.'
+            raise ValueError(error_str.format(self.mesh.name))
+
+        # this to have controlDict file for mesh definition
+        FoamFiles().write_default_files(self.path, 'simpleFoam',
+                                        self.time, False)
+
+        foamface.init(self.name, os.path.abspath(os.path.join(self.path,
+                                                              os.pardir)))
+
+    def tearDown(self):
+        if os.path.exists(self.test_dir):
+            shutil.rmtree(self.test_dir)
+
+    def test_add_mesh(self):
```

Comment:
Please see comment in https://github.com/simphony/simphony-openfoam/pull/10/files#r27658560, There should be something that can be checked regarding the operation of `add_mesh` and `write_mesh` that idealy does not invlove `get_mesh`.